network/openssl_cert.py

```diff
@@ -0,0 +1,315 @@
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.basic import *
+from OpenSSL import crypto
+from random import randint
+
+import os
+
+
+DOCUMENTATION = '''
+---
+module: openssl_cert
+author: "Yanis Guenane (@Spredzy)"
+version_added: "2.2"
+short_description: Generate OpenSSL Certificates
+description:
+    - "This module allows one to (re)generates OpenSSL certificates. It implements a notion
+       of provider (ie. 'self-signed', 'letsencrypt'), it can be further extended.
+       It uses the pyOpenSSL python library to interact with openssl."
+requirements:
+    - "python-pyOpenSSL"
+options:
+    state:
+        required: false
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the certificate should exist or not, , taking action if the state is different from what is stated.
+    name:
+        required: true
+        description:
+            - Name of the generated OpenSSL certificate
+    provider:
+        required: true
+        choices: ['self-signed', 'letsencrypt' ]
+        description:
+            -  Name of the provider to use to generate/retrieve the OpenSSL certificate
+    force:
+        required: false
+        default: False
+        choices: [ True, False ]
+        description:
+            - Should the certificate be forced regnerated by this ansible module
+    path:
+        required: true
+        description:
+            - Name of the folder in which the generated OpenSSL certificate  will be written
+    csr:
+        required: true
+        description:
+            - Path to the Certificate Signing Request (CSR) needed to generate this certificate
+
+    digest:
+        required: false
+        default: "sha256"
+        description:
+            - Digest used when signing the certifate with the private key
+    privatekey:
+        required: false
+        description:
+            - Path to the privatekey to use when signing the certificate
+    notBefore:
+        required: false
+        default: 0
+        description:
+            - Number of second from now which the generated certificate will be valid from
+    notAfter:
+        required: false
+        default: 315360000
+        description:
+            - Number of second from now which the generated certificate will be expired
+
+    accountkey:
+        required: false
+        description:
+            - Path to the accountkey
+    challenge:
+        required: false
+        description:
+            - Path to the ACME challenge directory
```

Comment:
This should be a directory that's served on `http://<HOST>:80/.well-known/acme-challenge/`, right?
It's not clear from this description.